Source/WebKit/WebProcess/GPU/graphics/cocoa/ImageBufferShareableMappedIOSurfaceBitmapBackend.cpp

```diff
@@ -74,15 +74,41 @@ ImageBufferBackendHandle ImageBufferShareableMappedIOSurfaceBitmapBackend::creat
 
 GraphicsContext& ImageBufferShareableMappedIOSurfaceBitmapBackend::context()
 {
-    if (!m_context) {
+    if (m_context) {
+        CGContextRef cgContext = m_context->platformContext();
+        if (m_lock || !cgContext) {
+            // The existing context is a valid context and the IOSurface is locked, or alternatively
+            // the existing context is an invalid context, for some reason we ran into an error previously.
+            return *m_context;
+        }
+
+        // The IOSurface is unlocked for every flush to prepare for external access by the compositor.
+        // Re-lock on first context() request after the external access has ended and new update starts.
+        if (auto lock = m_surface->lock(IOSurface::AccessMode::ReadWrite)) {
+            auto bitsPerComponent = m_surface->bitmapConfiguration().bitsPerComponent;
+            auto bitsPerPixel = bitsPerComponent * 4;
+            auto bytesPerRow = m_surface->bytesPerRow();
+            auto size = m_surface->size();
+            auto success = CGBitmapContextSetData(cgContext, 0, 0, size.width(), size.height(), lock->surfaceBaseAddress(), bitsPerComponent, bitsPerPixel, bytesPerRow);
```

Comment:
Is there a data copy under here, or is this just updating the context with a new base address?

Reply:
Updating the context with a new base address.